File: eukaryotes/constant_variables.R
###
### DESCRIPTION:
###
### This script contains some variables and functions which are used in many 
### different steps of the data acquisition process. Thus, they are concentrated 
### here to loaded thought the command source('constant_variables.R') wherever
### they are necessary. Some of the following variables are related with
### tsv files stored in the 'files' directory.
###
###

ensembl_parameters_df <- read.csv('./files/ensembl_parameters.tsv', sep='\t')
model_species_df <- read.csv('./files/model_species.tsv', sep='\t')
model_species <- as.character(model_species_df$species)

define_main_dir <- function(x) {
    paste('./data/',x,'/preprocessing/', sep='')
}

valid_keywords_for_hsp40 <- c('heat shock 40', 'heat shock protein 40', 
                              'heat shock cognate 40', 'shock 40kDa', 
                              'shock 40 kDa', 'chaperone 40', 'hsp40', 'dnaj')
valid_keywords_for_hsp70 <- c('heat shock 70', 'heat shock protein 70', 
                              'heat shock cognate 70', 'heat shock cognate 71',
                              'shock 70kDa', 'shock 70 kDa', 'shock 71 kDa',
                              'chaperone 70', 'hsp70', 'dnak')
prohibited_keywords <- c('interact', 'cooporate', 'stimulate', 'resemble', 
                         'contain', 'unknown function', 'with the hsp70', 
                         'with the hsp40', 'escort', 'fragment', 'pseudogene', 
                         'complex', 'sense overlapping', 'sense intronic', 
                         'antisense', 'binding', 'exchange factor',
                         'co-chaperone', 'suppressor', 'cooperates with',
                         'organizing protein', 'readthrough', 'divergent transcript')

File: eukaryotes/constant_variables.py
# -*- coding: utf-8 -*-
import pandas
import os


mongodb = 'background'
parameters_df = pandas.read_csv('./files/ensembl_parameters.tsv', sep='\t',
                                index_col=0)
ensembl_classes = parameters_df.loc[:, 'class'].tolist()


model_species_df = pandas.read_csv('./files/model_species.tsv', sep='\t', 
                                     index_col=0)
model_species = model_species_df.species.tolist()
model_species_classes = dict(zip(model_species_df.species, 
                                   model_species_df.loc[:,'class']))

                                 
define_main_dir = lambda x: os.path.abspath('./data/'+x+'/preprocessing/')+'/'
